kratos/solving_strategies/strategies/solving_strategy.h

```diff
@@ -129,14 +138,17 @@ class SolvingStrategy
     explicit SolvingStrategy(
         ModelPart& rModelPart,
         bool MoveMeshFlag = false
-        ) : mrModelPart(rModelPart)
+        ) : mpModelPart(&rModelPart)
     {
         SetMoveMeshFlag(MoveMeshFlag);
     }
 
     /** Destructor.
      */
-    virtual ~SolvingStrategy(){}
+    virtual ~SolvingStrategy()
+    {
+        mpModelPart =  nullptr;
```

Comment:
why is this needed?

Reply:
I don't trust compilers and its decisions, better make it explicit

Reply:
but it confuses others
This makes me think why is this necessary. Is there some special memory-related treatment necessary? (I know it is not but others reading might not now)

Reply:
can you explain your concerns? I have not faced such problems

Reply:
I prefer to do obvious code than ambiguous code that I know that one day the compiler will not compile due to criteria change. I am just compiler nihilist

Reply:
Is not needed because after the destructor call you are the `mpModelPart` member variable. So the value you assigned to it wouldn't affect anywhere.

Reply:
However, it is very important to set the mpModelPart to `nullptr` in the constructor if it is not initialized to avoid uninitialized pointer.

Reply:
Understood, I have already deleted it

Reply:
> However, it is very important to set the mpModelPart to `nullptr` in the constructor if it is not initialized to avoid uninitialized pointer.

Reverting the commit then

Reply:
It is already initialized in the variable declaration

Reply:
Can we proceed then? @pooyan-dadvand @philbucher